lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1591,11 +1631,15 @@ void HdVP2Mesh::_UpdateDrawItem(
                 _getColorData(_meshSharedData->_primvarInfo, colorArray, colorInterp);
                 _getOpacityData(_meshSharedData->_primvarInfo, alphaArray, alphaInterp);
 
-                if (colorInterp == HdInterpolationConstant
-                    && alphaInterp == HdInterpolationConstant) {
+                if ((colorInterp == HdInterpolationConstant
+                     || colorInterp == HdInterpolationInstance)
+                    && (alphaInterp == HdInterpolationConstant
+                        || alphaInterp == HdInterpolationInstance)) {
                     const GfVec3f& clr3f = UsdMayaColorSpace::ConvertLinearToMaya(colorArray[0]);
                     const MColor   color(clr3f[0], clr3f[1], clr3f[2], alphaArray[0]);
-                    shader = _delegate->GetFallbackShader(color);
+                    shader = _delegate->GetFallbackShader(
+                        color); // The color of the fallback shader is ignored when the
```

Comment:
Having the comment after the code breaks the line into two. I think it is better to have the comment in a separate line.